### DISCUSSION TITLE
Why gallery refreshes very slow at new items?

### DISCUSSION THREAD

**Discussion by rafnov** [FossifyOrg/General-Discussion#388] [Q&A]
Is it because of version 1.2.1 that has grown in size 2,5x since previous versions?
How do I make it faster?

**Comment by Aga-C** [comment]
Was it working faster in 1.1.3?

**Reply by rafnov** [reply]
Just downgraded and... Yes. It is significantly faster. My test procedure is the same:
1. Open gallery with folder view showing them miniatures of last images. 
2. Go to camera and make the photo.
3. Go back to backgrounded gallery.
In 1.1.3 new image in a folder loads in a second.
In 1.2.1 it loads few seconds and sometimes I feel it does nothing.

Btw, why gallery is so big now? Compatibility with many APIs?
For now I am ignoring 1.2.1 upgrade in Droid-ify.

**Reply by Aga-C** [reply]
We have a bug report for it, created at times of 1.1.0. Please add your remarks there: https://github.com/FossifyOrg/Gallery/issues/54.

> why gallery is so big now? Compatibility with many APIs?

The app size is bigger because we've added support for JPEG XL.

**Reply by rafnov** [reply]
Sorry, slow loading thumbnails is not the same issue as slow loading of new files and it does not depend on the amount of files in the folder here. I have one big DCIM/Camera, containig now over 4000 few MB files and few other folders like Screenshots, Messenger or other private stuff, containing dosen or few dosens of smaller files. And they all seem behaving the same.
Just made the screenshot of opened gallery main window and the file appeared there after two seconds in the folder of just 1 dosen of files. 1.1.3.

**Reply by Aga-C** [reply]
In this report there is:

> Sometimes if you take photos with a camera, or download them, they don't always show up instantly. They may need a minute or a pull to refresh to appear.

what sounds like the same as what you described above.

**Reply by rafnov** [reply]
Yes, you are right, sorry I missed that.
